### PR TITLE
Enhance ECD and refresh reqid for each use of C API

### PR DESCRIPTION
**Pull request by murlock**
##### SUMMARY

Enhance ECD and refresh `reqid for each use of C API.

The ECD did not use the reqid provided in each request, making difficult to investigate error 500. Please note only the EC upload/download has been patched.

On same trend, the C API was using an unique reqid for entire lifetime of `struct oio_sds_s *`, provided by `oio_sds_init`, even if several call of C API was made (`oio_sds_upload_from_buffer`, `oio_sds_upload`, ...). It was impossible to distinguish calls between them.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
C API, ECD

##### SDS VERSION
```
openio 5.6.1.dev2
```

### CODE REVIEW

File: oio/ecd/app.py [hunk 1]
```diff
@@ -254,7 +259,7 @@ def _on_metachunk_GET(self, req):
             meta_chunk[0]['size'] = \
                 int(safe_get_header(req, 'chunk_size'))
             return self.read_ec_meta_chunk(storage_method, meta_chunk,
-                                           my_range[0], my_range[1])
+                                           my_range[0], my_range[1], reqid)
```
Comment (AymericDu) [inline oio/ecd/app.py:262]:
```suggestion
                                           my_range[0], my_range[1], reqid=reqid)
```

File: oio/ecd/app.py [hunk 2]
```diff
@@ -210,14 +213,15 @@ def _on_metachunk_PUT(self, req):
...
+        reqid = req.headers.get(REQID_HEADER, request_id("ECD-"))
 
         if storage_method.ec:
             nb_chunks = (storage_method.ec_nb_data +
                          storage_method.ec_nb_parity)
             pos = safe_get_header(req, 'chunk_pos')
             meta_chunk = load_meta_chunk(req, nb_chunks, pos)
             return self.write_ec_meta_chunk(source, size, storage_method,
-                                            sysmeta, meta_chunk)
+                                            sysmeta, meta_chunk, reqid)
```
Comment (AymericDu) [inline oio/ecd/app.py:224]:
```suggestion
                                            sysmeta, meta_chunk, reqid=reqid)
```

File: core/sds.c [hunk 3]
```diff
@@ -1138,7 +1138,7 @@ oio_sds_download (struct oio_sds_s *sds, struct oio_sds_dl_src_s *dl,
 {
 	if (!sds || !dl || !snk || !dl->url)
 		return (struct oio_error_s*) BADREQ("Missing argument");
-	oio_ext_set_reqid (sds->session_id);
+	oio_ext_set_prefixed_random_reqid("DL-");
```
Comment (fvennetier) [inline core/sds.c:1141]: If we do that, `sds->session_id` is of no use. Can't we set a random request ID only if `sds->session_id` is NULL or empty?

Comment (murlock) [reply core/sds.c:1141]: `session_id` is always initialized in `oio_sds_init` but since it is an opaque struct, we can remove it.

Comment (fvennetier) [reply core/sds.c:1138]: I don't like the idea of removing a useful feature but ¯\_(ツ)_/¯

**Review by fvennetier — APPROVED** [review]

**Review by AymericDu — APPROVED** [review]

**Pull request merged**